sphynx/r/derive_table.r

```diff
@@ -0,0 +1,23 @@
+# Run user code on a table.
+source("r/util.r")
+
+# Create input tables.
+for (t in names(inputs)) {
+    assign(t, input_parquet(t))
+}
+
+# Run user code.
+print("RUNNING USER CODE")  # For log cleanup.
+code <- params[["code"]]
+eval(parse(text = code))
+print("USER CODE FINISHED")
+
+# Save outputs.
+print(params[["outputFields"]])
+for (t in names(outputs)) {
+    print(t)
+    fields <- params[["outputFields"]]
+    fs <- fields[fields$parent == t, ]
```

Comment:
> Please comment on why the comma is needed here.

Looks like this is just basic R. See https://stackoverflow.com/questions/1686569/filter-data-frame-rows-by-a-logical-condition.

I like the `filter(df, ...)` syntax more so I tried switching to that. `fs <- filter(fields, parent == parent)` But here the condition is always true! Filter has no way of knowing I was thinking of the column on the left side and the local variable on the right side!

That brings me to https://stackoverflow.com/questions/40169949/filter-dataframe-using-global-variable-with-the-same-name-as-column-name which says _"just use `df[df$y==y,]` instead of `dplyr`."_

I guess when we get more familiar with R it won't be an issue.